shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/viewModel/ErrorBannerViewModel.kt

```diff
@@ -0,0 +1,104 @@
+package com.mbta.tid.mbta_app.viewModel
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.mbta.tid.mbta_app.model.ErrorBannerState
+import com.mbta.tid.mbta_app.repositories.IErrorBannerStateRepository
+import com.mbta.tid.mbta_app.utils.EasternTimeInstant
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+
+public interface IErrorBannerViewModel {
+    public val models: StateFlow<ErrorBannerViewModel.State>
+
+    public fun clearState()
+
+    public fun setIsLoadingWhenPredictionsStale(isLoading: Boolean)
+
+    public fun checkPredictionsStale(
+        predictionsLastUpdated: EasternTimeInstant,
+        predictionQuantity: Int,
+        action: () -> Unit,
+    )
+}
+
+public class ErrorBannerViewModel(private val errorRepository: IErrorBannerStateRepository) :
+    MoleculeViewModel<ErrorBannerViewModel.Event, ErrorBannerViewModel.State>(),
+    IErrorBannerViewModel {
+
+    public data class State(
+        val loadingWhenPredictionsStale: Boolean = false,
+        val errorState: ErrorBannerState? = null,
+    )
+
+    public sealed interface Event {
+        public data class SetIsLoadingWhenPredictionsStale(val isLoading: Boolean) : Event
+
+        public data object ClearState : Event
+    }
+
+    @Composable
+    override fun runLogic(events: Flow<Event>): State {
+        var awaitingPredictionsAfterBackground: Boolean by remember { mutableStateOf(false) }
+        var errorState: ErrorBannerState? by remember { mutableStateOf(null) }
+
+        LaunchedEffect(Unit) {
+            errorRepository.subscribeToNetworkStatusChanges()
+            errorRepository.state.collect { errorState = it }
```

Comment:
It almost seems like the ErrorRepository could be removed and entirely replaced in favor of this VM... but that felt like a bigger change than I wanted to make right now.